src/firebolt/model/engine.py

```diff
@@ -320,6 +325,95 @@ def stop(
 
         return engine
 
+    def update(
+        self,
+        name: Optional[str] = None,
+        engine_type: Optional[EngineType] = None,
+        scale: Optional[int] = None,
+        spec: Optional[str] = None,
+        auto_stop: Optional[int] = None,
+        warmup: Optional[WarmupMethod] = None,
+        description: Optional[str] = None,
+    ) -> Engine:
+        """
+        update the engine, and returns an updated version of the engine, all parameters
+        could be set to None, this would keep the old engine parameter value
+        """
+
+        class _EngineUpdateRequest(FireboltBaseModel):
```

Comment:
Same question as for a database update PR: can we have a plain dict here?

Reply:
Generally it is possible, but the code will get a bit uglier, since, we would need to call `.jsonable_dict()` for the each dict field, which is `Object` and additionally check whether it is not `None`. Currently, it is all done automatically within `FireboltBaseModel`